File: backend/src/domain/repositories/IHelpRequestRepository.ts
import { HelpRequest, HelpRequestStatus, HelpRequestType } from '../entities/HelpRequest';

export interface IHelpRequestRepository {
  findById(id: string): Promise<HelpRequest | null>;
  findByEmergency(emergencyId: string): Promise<HelpRequest[]>;
  findByRequester(requesterId: string): Promise<HelpRequest[]>;
  findOpenByEmergency(emergencyId: string): Promise<HelpRequest[]>;
  findByType(emergencyId: string, type: HelpRequestType): Promise<HelpRequest[]>;
  findByStatus(emergencyId: string, status: HelpRequestStatus): Promise<HelpRequest[]>;
  findUrgent(emergencyId: string): Promise<HelpRequest[]>;
  save(request: HelpRequest): Promise<HelpRequest>;
  update(request: HelpRequest): Promise<HelpRequest>;
  delete(id: string): Promise<void>;
}

File: backend-dotnet/src/Hestia.Domain/Repositories/IHelpRequestRepository.cs
using Hestia.Domain.Entities;
using Hestia.Domain.Enums;

namespace Hestia.Domain.Repositories;

public interface IHelpRequestRepository
{
    Task<HelpRequest?> FindByIdAsync(Guid id, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindByEmergencyAsync(Guid emergencyId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindByRequesterAsync(Guid requesterId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindOpenByEmergencyAsync(Guid emergencyId, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindByTypeAsync(Guid emergencyId, HelpType type, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindByStatusAsync(Guid emergencyId, HelpRequestStatus status, CancellationToken ct = default);
    Task<IReadOnlyList<HelpRequest>> FindUrgentAsync(Guid emergencyId, CancellationToken ct = default);
    Task<HelpRequest> SaveAsync(HelpRequest request, CancellationToken ct = default);
    Task<HelpRequest> UpdateAsync(HelpRequest request, CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}
